mlflow/insights/models/entities.py

```diff
@@ -0,0 +1,364 @@
+"""
+MLflow Insights entities.
+
+Main entity models for the MLflow Insights agent that analyzes traces to discover
+and document issues with AI agents. Supports AI-guided investigation workflows
+with hypotheses, evidence collection, and issue tracking.
+"""
+
+from __future__ import annotations
+
+from typing import Annotated, Any
+from uuid import uuid4
+
+from pydantic import AfterValidator, Field, field_validator
+
+from mlflow.exceptions import MlflowException
+from mlflow.insights.constants import AnalysisStatus, HypothesisStatus, IssueSeverity, IssueStatus
+from mlflow.insights.models.base import (
+    EvidencedModel,
+    EvidenceEntry,
+    ExtensibleModel,
+    SerializableModel,
+    TimestampedModel,
+)
+from mlflow.insights.utils import normalize_evidence
+
+
+def validate_non_empty_string(v: str, field_name: str) -> str:
```

Comment:
Do we need all those definitions in this file? Could we move them to utils?

Reply:
The Annotators definitely belong here as they're related to validators for these models. The actual functions that are being called have been moved to utils.